### PR TITLE
feat: add timeout option for both install and test phase

### PR DESCRIPTION
**Pull request by SimenB**
##### Checklist

- [x] `npm test` passes
- [ ] tests are included
- [x] documentation is changed or added
- [x] contribution guidelines followed
      [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)

This is #740 without the changes to lookups. I have the same question @Zirro had over there though - where to add tests? Seems like a bunch of the options you can define in `lookup.json` are untested.

### CODE REVIEW

File: lib/common-args.js [hunk 1]
```diff
@@ -73,8 +73,7 @@ module.exports = function commonArgs(app) {
     .option('timeout', {
       alias: 'o',
       type: 'number',
-      description: 'Set timeout for npm install',
-      default: 1000 * 60 * 10
+      description: 'Set timeout for npm install'
```
Comment (SimenB) [inline lib/common-args.js]: we use it for `test` now - maybe a dedicated `test` timeout makes sense?

**Comment by SimenB** [comment]
Would love to make some progress here as well - any feedback on it? 🙂

**Comment by SimenB** [comment]
I've rebased now. Happy to add tests if I could get some guidance on how to best test this

**Comment by targos** [comment]
I'm always lost with tests in this repo. I wouldn't be against landing this without new tests.

**Comment by SimenB** [comment]
Feel free to merge 😃

**Comment by targos** [comment]
I cannot merge a draft PR :p

**Comment by SimenB** [comment]
It's not draft anymore 😀

File: README.md [hunk 2]
```diff
@@ -143,6 +143,7 @@ For syntax, see [lookup.json](./lib/lookup.json), the available attributes are:
 "useGitClone": true          Use a shallow git clone instead of downloading the module
 "ignoreGitHead":             Ignore the gitHead field if it exists and fallback to using github tags
 "yarn":                      Install and test the project using yarn instead of npm
+"timeoutLength":             Number of milliseconds before timeout
```
Comment (targos) [inline README.md:146]:
```suggestion
"testTimeout":             Number of milliseconds before timeout
```

Thinking about this more... If this is used only for tests, maybe `testTimeout` is a better name here?

**Comment by SimenB** [comment]
@targos I was a bit quick there, need to update the code as well. I'll get back to this later tonight

File: lib/lookup.js [hunk 3]
```diff
@@ -162,6 +162,9 @@ function resolve(context) {
       if (rep.yarn) {
         context.module.useYarn = true;
       }
+      if (rep.timeoutLength) {
+        context.module.timeoutLength = rep.timeoutLength;
```
Comment (targos) [inline lib/lookup.js:166]:
```suggestion
      if (rep.testTimeout) {
        context.module.testTimeout = rep.testTimeout;
```

File: lib/package-manager/test.js [hunk 4]
```diff
@@ -104,7 +104,14 @@ async function test(packageManager, context) {
...
 
       const proc = spawn(bin, args, options);
-      const finish = timeout(context, proc, runScript, 'Test');
+      const finish = timeout(
+        packageManager,
+        context,
+        proc,
+        runScript,
+        'Test',
+        context.module.timeoutLength
```
Comment (targos) [inline lib/package-manager/test.js:113]:
```suggestion
        context.module.testTimeout
```

**Comment by targos** [comment]
ping @SimenB :)

**Comment by SimenB** [comment]
@targos sorry, completely forgot. Made a search & replace pass now, PTAL 🙂

File: README.md [hunk 5]
```diff
@@ -143,6 +143,7 @@ For syntax, see [lookup.json](./lib/lookup.json), the available attributes are:
 "useGitClone": true          Use a shallow git clone instead of downloading the module
 "ignoreGitHead":             Ignore the gitHead field if it exists and fallback to using github tags
 "yarn":                      Install and test the project using yarn instead of npm
+"testTimeout":               Number of milliseconds before timeout
```
Comment (SimenB) [inline README.md:146]: @targos what about just `timeout` to match the CLI arg? It applies to both `install` and `test`

**Comment by targos** [comment]
I think we shouldn't rename everything to `testTimeout` though because the general option also applies to install.

File: README.md (hunk 5 above)
Comment (targos) [reply README.md:146]: I agree with that.

Comment (SimenB) [reply README.md]: done

**Comment by SimenB** [comment]
@targos what about now? I quite like it as it brings parity between option from `lookup` and CLI arg.

**Comment by targos** [comment]
Looks good to me. Thanks, let's ship it!

**Comment by SimenB** [comment]
Yay!

**Pull request merged**